### DISCUSSION TITLE
DXF Import

### DISCUSSION THREAD

**Discussion by CarbonFlora** [hannobraun/fornjot#1977] [Q&A] [answered]
I've taken a pot shot at implementing import functionality for the DXF file format. I'm trying to pass through a Handle<Cycle> into handle_model(). To create a cycle, you create a vec of Handle<HalfEdge>. I am stuck at trying to create a HalfEdge. More specifically, how do I create a boundary for HalfEdge? I'm not perfect on interpreting Rust syntax, but anything will help. Neat CAD kernel btw.

**Comment by hannobraun** [comment, marked as answer]
Hey @CarbonFlora, thank you for your interest!


> I am stuck at trying to create a HalfEdge. More specifically, how do I create a boundary for HalfEdge?

The best way to create a `HalfEdge` is probably the [`BuildHalfEdge`](https://github.com/hannobraun/fornjot/blob/main/crates/fj-core/src/operations/build/edge.rs) trait. If you `use` that trait, you can just call its methods on `HalfEdge`, for example:

``` rust
use fj_core::operations::BuildHalfEdge;
HalfEdge::line_segment([[0., 0.], [1., 0.]], None, services);
```

There are more of these builder traits. You can build a whole [`Cycle`](https://github.com/hannobraun/fornjot/blob/main/crates/fj-core/src/operations/build/cycle.rs) or [`Region`](https://github.com/hannobraun/fornjot/blob/main/crates/fj-core/src/operations/build/region.rs), for example.

Hope that helps!


> Neat CAD kernel btw.

Thank you!
